src/object/cli_shard.c

```diff
@@ -1125,7 +1227,9 @@ dc_obj_shard_rw(struct dc_obj_shard *shard, enum obj_rpc_opc opc,
 	rw_args.co = shard->do_co;
 	rw_args.shard_args = args;
 	/* remember the sgl to copyout the data inline for fetch */
-	rw_args.rwaa_sgls = (opc == DAOS_OBJ_RPC_FETCH) ? sgls : NULL;
+	rw_args.rwaa_sgls = sgls;
+	rw_args.send_time = daos_client_metric ? daos_get_ntime() : 0;
+	obj_shard_update_metrics_begin(req);
```

Comment:
this is better, but still i'd rather have:
if (daos_client_metrics) {
  rw_args.send_time = daos_get_ntime();
  obj_shard_update_metrics_begin(req);
} else {
  rw_args.send_time = 0
}
and same for the obj_shard_update_metrics_end() calls.
i know we currently return immediately from those calls, but this is probably cleaner / safer for future changes that might go into those funcitons.

Reply:
it's fine unless you need to repush which looks like you might need to anyway because of a merge conflict